File: frontend/src/types/game.ts
export interface Seat {
  index: number
  display_name: string
  stack: number
  status: string
  current_bet: number
  total_hand_bet: number
  is_hero: boolean
  is_bot: boolean
  hole_cards: string[]
}

export interface TableState {
  table_id: string
  seats: Seat[]
  community_cards: string[]
  pot: number
  button_index: number
  current_street: number
  active_seat_index: number
  hand_number: number
  is_hand_in_progress: boolean
  hero_seat_index: number
  legal_actions: string[]
  waiting_for_hero: boolean
  hand_complete: boolean
  winner: { seat_index: number; display_name: string; reason: string } | null
  showdown: boolean
  pot_results: Array<{
    pot_amount: number
    winners: Array<{ seat_index: number; display_name: string }>
    winning_hand: string
  }>
}

export const STREET_NAMES = ['Preflop', 'Flop', 'Turn', 'River', 'Showdown']

File: backend/app/schemas/game.py
from pydantic import BaseModel


class CreateTableRequest(BaseModel):
    bot_styles: list[str] | None = None


class ActionRequest(BaseModel):
    action_type: str  # fold, check, call, raise, all_in
    amount: int | None = None


class TableResponse(BaseModel):
    table_id: str
    seats: list[dict]
    community_cards: list[str]
    pot: int
    button_index: int
    current_street: int
    active_seat_index: int
    hand_number: int
    is_hand_in_progress: bool
    hero_seat_index: int
    legal_actions: list[str] = []
    waiting_for_hero: bool = False
    hand_complete: bool = False
    winner: dict | None = None
    showdown: bool = False
    pot_results: list[dict] = []
